### PR TITLE
Add sort and fix where on _id field

### PR DESCRIPTION
**Pull request by orlichter1**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:https://github.com/demisto/etc/issues/28014

## Description
Add sort and fix where on _id field

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 1]
```diff
@@ -161,7 +163,15 @@ def convert_id_to_object_id(
...
         if '_id' in entry:
-            entry['_id'] = ObjectId(entry['_id'])
+            id_data = entry['_id']
+            if isinstance(id_data, str):
+                entry['_id'] = ObjectId(id_data)
+            if isinstance(id_data, dict):
+                new_dict = {}
+                for key, value in id_data.items():
+                    new_dict[key] = ObjectId(value)
+                entry['_id'] = new_dict
```
Comment (yaakovi) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:173]:
```suggestion
                entry['_id'] = dict((key, ObjectId(value)) for new_dict)
```

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 2]
```diff
@@ -316,6 +325,29 @@ def validate_json_objects(json_obj: Union[dict, list]) -> Union[dict, list]:
...
+
+    Returns:
+        list accepted by pymongo.sort()
+    """
+    sort_fields = sort_str.split(',')
+    sort_list = list()
+    for field in sort_fields:
+        if ':' not in field:
+            raise ValueError("`sort` is not in the correct format.")
+        field, type = field.split(':')
```
Comment (yaakovi) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:343]:
```suggestion
        field, type = field.split(':', 1)
```

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 3]
```diff
@@ -316,6 +325,29 @@ def validate_json_objects(json_obj: Union[dict, list]) -> Union[dict, list]:
...
+        if type not in SORT_TYPE_DICT.keys():
```
Comment (yaakovi) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:344]:
```suggestion
        if type not in SORT_TYPE_DICT:
```

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 4]
```diff
@@ -316,6 +325,29 @@ def validate_json_objects(json_obj: Union[dict, list]) -> Union[dict, list]:
...
+            raise ValueError("`sort` is not in the correct format. Please make sure it's either 'asc' or 'desc'")
+        else:
```
Comment (yaakovi) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:346]: no need for `else` after raise

File: Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py [hunk 5]
```diff
@@ -196,3 +200,26 @@ def test_query(mocker):
...
+
+class TestFormatSort:
+    def test_format_sort_correctly(self):
+        """
+        Given:
+            a sort string in the correct format
+        Then:
+            Format the string in the correct format to be used in `pymongo.sort()`
+        """
+        assert format_sort("field1:asc,field2:desc") == [('field1', 1), ('field2', -1)]
```
Comment (yaakovi) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py:213]: add a test with a single field too

**Review by yaakovi — APPROVED** [review]

**Comment by kirbles19** [comment]
@orlichter1 Updated the RN, everything else looked fine.

**Pull request merged**